op-batcher/batcher/service.go

```diff
@@ -125,9 +126,16 @@ func (bs *BatcherService) initRPCClients(ctx context.Context, cfg *CLIConfig) er
 	}
 	bs.L1Client = l1Client
 
-	endpointProvider, err := dial.NewStaticL2EndpointProvider(ctx, bs.Log, cfg.L2EthRpc, cfg.RollupRpc)
+	var endpointProvider dial.L2EndpointProvider
+	if strings.Contains(cfg.RollupRpc, ",") || strings.Contains(cfg.L2EthRpc, ",") {
```

Comment:
Shouldn't this be 
```suggestion
	if strings.Contains(cfg.RollupRpc, ",") && strings.Contains(cfg.L2EthRpc, ",") {
```
I mean, if only one is a list, it's a wrong config anyways. Could also add a check to `CLIConfig.Check` that both are lists of the same length.

Reply:
My thinking was that if just one list is provided, we should avoid silently falling back to the `Static` provider. Adding the check to `CLIConfig.Check` is a better way of achieving this. :+1: (Edit: there, though, the check _will_ be `||` so that it will length-check and then error in the just-one-list provided case.)